modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java

```diff
@@ -366,6 +374,24 @@ public static HttpResponse sendGetRequest(String url, String userAgent, HttpClie
         return httpClient.execute(request);
     }
 
+    public static HttpResponse sendECPPostRequest(String url, String userAgent, HttpClient httpClient, String username, String password, String soapRequest) {
+        HttpPost request = new HttpPost(url);
+        HttpResponse response = null;
+        String auth = username + ":" + password;
+        byte[] encodedAuth = Base64.encodeBase64(
+                auth.getBytes(StandardCharsets.UTF_8));
+        String authHeader = "Basic " + new String(encodedAuth);
+        request.setHeader(HttpHeaders.AUTHORIZATION, authHeader);
+        request.setHeader(HttpHeaders.CONTENT_TYPE, "text/xml; charset=utf-8");
+        try {
+            request.setEntity(new StringEntity(soapRequest));
+            response = httpClient.execute(request);
+        } catch (Exception e) {
+            log.error("Error while sending the ECP Request");
```

Comment:
If an exception occurs, the response will be equal to null. So the subsequent calls will be failed with NullPointer Exceptions.

Reply:
> If an exception occurs, the response will be equal to null. So the subsequent calls will be failed with NullPointer Exceptions.

fixed